plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -912,6 +917,23 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
 
         boolean external = true;
         String location = getLocation(tableMetadata.getProperties());
+        ConnectorTableHandle connectorTableHandle = getTableHandle(session, tableMetadata.getTable());
```

Comment:
Add an assertion into `io.trino.plugin.deltalake.TestDeltaLakeBasic#testCorruptedTableLocation`

It may be a straightforward user behavior that when the table is corrupted to attempt to "replace" it.